### PR TITLE
Fix disconnected support due to support horizontal expansion

### PR DESCRIPTION
**Pull request by rijkvanmanen**
By checking of the model is splitting the support areas, this undesired behaviour can be easily identified. By expanding the model with the same amount as the horizontal expansion, it can be found which part(s) need to be removed.

CURA-9844

### CODE REVIEW

File: src/support.cpp [hunk 1]
```diff
@@ -1013,7 +1013,25 @@ void AreaSupport::generateSupportAreasForMesh(SliceDataStorage& storage,
...
+                // the support areas in this layer are split into parts by the model itself
+                int i = 0;
+                for (PolygonsPart poly : layer_this.splitIntoParts())
+                {
+                    if (poly.intersection(layer_this_without_expansion).empty())
+                    {
+                        // this area is expanded to the other side of the model and not connected to the original area that needed support
+                        layer_this.remove(i);
+                    }
+                    i++;
```
Comment (smartavionics) [inline src/support.cpp:1032]: Do you really want to increment i when you have just removed the the i'th element ?

File: src/support.cpp [hunk 2]
```diff
@@ -1013,7 +1013,25 @@ void AreaSupport::generateSupportAreasForMesh(SliceDataStorage& storage,
                 layer_above = &empty;
                 layer_this = layer_this.unionPolygons(storage.support.supportLayers[layer_idx].support_mesh);
             }
-            layer_this = AreaSupport::join(storage, *layer_above, layer_this, smoothing_distance).difference(model_mesh_on_layer);
+            layer_this = AreaSupport::join(storage, *layer_above, layer_this, smoothing_distance);
+            const int layer_size = layer_this.size();
```
Comment (Joeydelarago) [inline src/support.cpp:1017]: You should give this a more descriptive name, it isn't clear a few lines down what the difference is between layer\_size and layer\_this.size()

File: src/support.cpp [hunk 3]
```diff
@@ -1013,7 +1013,25 @@ void AreaSupport::generateSupportAreasForMesh(SliceDataStorage& storage,
...
+
+            Polygons layer_this_without_expansion = layer_this.difference(model_mesh_on_layer.offset(extension_offset));
+            layer_this = layer_this.difference(model_mesh_on_layer);
+            if (layer_size < layer_this.size())
+            {
+                // the support areas in this layer are split into parts by the model itself
+                int i = 0;
+                for (PolygonsPart poly : layer_this.splitIntoParts())
```
Comment (Joeydelarago) [inline src/support.cpp:1025]: This could also do with a more descriptive name. Something like layer\_part or support\_part

**Review by Joeydelarago — CHANGES_REQUESTED** [review]

**Comment by rijkvanmanen** [comment]
A different solution for the same problem is proposed in CURA-9996. I will come back to this PR after we finished the work for CURA-9996.

**Comment by rijkvanmanen** [comment]
A better solution is now implemented with CURA-10152. So this PR will be closed

**Pull request closed without merging**